src/main/java/seedu/address/ui/CalendarPanel.java

```diff
@@ -0,0 +1,341 @@
+        GregorianCalendar gregorianCalendar = new GregorianCalendar(year, month, 1);
+        int leadGap = calculateLeadGap(gregorianCalendar);
+        int daysInMonth = calculateDaysInMonth(gregorianCalendar);
+        int tailGap = calculateTailGap(daysInMonth + leadGap);
+
+        String[] before = previousMonth(month, leadGap);
+        String[] current = currMonth(daysInMonth);
+        String[] after = nextMonth(month, tailGap);
+
+        String[] combined = combine(before, current, after);
+
+        fillUpGrid(before.length, current.length, combined);
+    }
+
+    /**
+     * Calculates excess grid boxes at the front of the calendar.
+     * @param gc Calendar containing selected date
+     * @return int Number of excess grid boxes.
+     */
+    private int calculateLeadGap(GregorianCalendar gc) {
+        int result = gc.get(Calendar.DAY_OF_WEEK) - 1;
+        if (result == 0) {
+            result = 7;
+        }
+        return result;
+    }
+
+    /**
+     * Calculates number of days in selected date.
+     * @param gc Calendar containing selected date
+     * @return int Number of days in selected date.
+     */
+    private int calculateDaysInMonth(GregorianCalendar gc) {
+        int result = DAYS_IN_MONTH[month];
+        if (gc.isLeapYear(gc.get(Calendar.YEAR)) && month == 1) {
+            result++;
+        }
+        return result;
+    }
+
+    /**
+     * Calculates excess grid boxes at the end of the calendar.
+     * @param n Number of days of calendar already filled up
+     * @return int Number of excess grid boxes.
+     */
+    private int calculateTailGap(int n) {
+        return 42 - n;
+    }
+
+    /**
+     * Generates array containing days from previous month.
+     * @param currMonth Current selected month
+     * @param leadGap Number of days needed from previous month
+     * @return String array containing days from previous month.
+     */
+    private String[] previousMonth(int currMonth, int leadGap) {
+        Calendar c = Calendar.getInstance();
+        c.set(Calendar.YEAR, year);
+        c.set(Calendar.MONTH, currMonth);
+        c.add(Calendar.MONTH, -1);
+        int y = c.get(Calendar.YEAR);
+        int m = c.get(Calendar.MONTH);
+
+        GregorianCalendar gc = new GregorianCalendar(y, m, 1);
+        int daysInMonth = DAYS_IN_MONTH[m];
+        if (gc.isLeapYear(gc.get(Calendar.YEAR)) && m == 1) {
+            daysInMonth++;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (int i = daysInMonth; i > 0; i--) {
+            sb.append(i).append(" ");
+        }
+        String[] allDays = sb.toString().split("\\s");
+        String[] neededDays = Arrays.copyOfRange(allDays, 0, leadGap);
+        neededDays = reverse(neededDays);
+        return neededDays;
+    }
+
+    /**
+     * Reverses a String array.
+     * @param s String array to be reversed
+     * @return Reversed String array.
+     */
+    private String[] reverse(String[] s) {
```

Comment:
I assume this is populating the days of the previous month from the back? Good work!

Reply:
yup! it's an innovative approach!